### DISCUSSION TITLE
For Beginners

### DISCUSSION THREAD

**Discussion by louperson** [amazon-science/chronos-forecasting#88] [General]
Is there a guide on how to get started with Chronos on AWS sagemaker studio?  I have pretty good AWS skills but just need a very simple guide on how to get started.  I just need that first "hello world" break through to piece together an example stack.  Thank you!

**Comment by lostella** [comment]
@louperson what did you try that did not work? Do the examples from the [README](https://github.com/amazon-science/chronos-forecasting?tab=readme-ov-file#-usage) not work out of the box there? You should be able to pip install the inference code as described there, also on SageMaker Studio notebooks I suppose.

**Comment by louperson** [comment]
Thank you very much. I did that. I got that far. Then I tried loading the
synthetic time series data but that did not complete. I’d like to do a
simple blog showing how to build an application in stage maker studio using
this model in a lab format for other beginners like me.

**Reply by lostella** [reply]
What happens when trying to load the synthetic data? If you could post a snippet of the code and error you got, we can probably figure out what's going on

**Comment by louperson** [comment]
Thanks for help, I successfully completed these steps but got stuck on step 7, any ideas?
1) Logged into AWS Console.
2) Went to Amazon Sagemaker.
3) Created a notebook instance.  Made sure to choose a GPU instance.
4) Got into a JupyterLab container.  Chose other and terminal.
5) Followed the Read.me, installed Chronos pip install git+https://github.com/amazon-science/chronos-forecasting.git
Confirmed it completed successfully.  "Successfully built chronos"
6) In the container, created a file with this command: nano lab.py and then combined the 2 code examples in the read.me which you can see in APPENDIX 1 below.  
7) When I execute the code python lab.by  I combined both code snippets.  I don't get any output when the plot commands run?  Any ideas?

APPENDIX 1

import pandas as pd  # requires: pip install pandas
import matplotlib.pyplot as plt  # requires: pip install matplotlib
import numpy as np
import torch
from chronos import ChronosPipeline

pipeline = ChronosPipeline.from_pretrained(
    "amazon/chronos-t5-small",
    device_map="cuda",  # use "cpu" for CPU inference and "mps" for Apple Silicon
    torch_dtype=torch.bfloat16,
)

df = pd.read_csv("https://raw.githubusercontent.com/AileenNielsen/TimeSeriesAnalysisWithPython/master/data/AirPassengers.csv")

forecast = pipeline.predict(
    context=torch.tensor(df["#Passengers"]),
    prediction_length=12,
    num_samples=20,
)

forecast_index = range(len(df), len(df) + 12)
print(forecast_index)
low, median, high = np.quantile(forecast[0].numpy(), [0.1, 0.5, 0.9], axis=0)

plt.figure(figsize=(8, 4))
plt.plot(df["#Passengers"], color="royalblue", label="historical data")
plt.plot(forecast_index, median, color="tomato", label="median forecast")
plt.fill_between(forecast_index, low, high, color="tomato", alpha=0.3, label="80% prediction interval")
plt.legend()
plt.grid()
plt.show()

print(ChronosPipeline.predict.__doc__)

**Reply by louperson** [reply]
Being stuck at step 7, it has something to do with Jupyter lab and the terminal and matlibplot display.  I was able to get everything working as a Jupyter notebook.  If I figure it out, I close out there in case someone else has the problem.

**Comment by louperson** [comment]
Got a hello world application working using Jupyter Notebook (still want to get it working using Jupyter Lab) as laid out in the Read.me
1) Logged into AWS Console.
2) Went to Amazon Sagemaker.
3) Notebook-Notebook Instances.
4) Create notebook instance.
5) Under IAM Role, selected create role and used AmazonSagemaker-ExecutionRole.
6) Important and obvious: Select a GPU instance type that is supported by the selected platform identifier. 
7) Click Open Jupyter.
8) Run the PIP command in the Read.me to install Chronos.
9) Copy and paste the code under Forecasting into a cell and hit Run.
10) The copy and paste the code under "We can now visualize the forecast:" and hit run.
![git1](https://github.com/amazon-science/chronos-forecasting/assets/20251531/77cd3d14-dc8a-441e-abfa-2c32f25b1635)
![image](https://github.com/amazon-science/chronos-forecasting/assets/20251531/0d412929-67a8-4db7-89ea-573b39fb5ccc)

**Comment by louperson** [comment]
Thanks, I'll check it out!
